### PR TITLE
ondemand: only select allowed node types for capture

### PR DESCRIPTION
**Pull request by lebauce**

### CODE REVIEW

**Comment by lebauce** [comment]
run skydive-functional-hw-tests

**Comment by lebauce** [comment]
run skydive-cdd-overview-tests
run skydive-functional-tests-backend-orientdb
run skydive-scale-tests

**Comment by lebauce** [comment]
run skydive-functional-tests-backend-orientdb

**Comment by lebauce** [comment]
run skydive-functional-tests-backend-orientdb

File: flow/ondemand/client/client.go [hunk 1]
```diff
@@ -96,6 +97,6 @@ func NewOnDemandFlowProbeClient(g *graph.Graph, ch api.Handler, agentPool ws.Str
 		nodeTypes[i] = nodeType
 		i++
 	}
-	nodeTypeQuery := new(gremlin.QueryString).Has("Type", gremlin.Within(nodeTypes...)).String()
+	nodeTypeQuery := new(gremlin.QueryString).Has("Host", gremlin.Ne("")).Has("Type", gremlin.Within(nodeTypes...)).String()
```
Comment (safchain) [inline flow/ondemand/client/client.go]: why not Has("Host", gremlin.Ne(""), "Type", gremlin.Within(nodeTypes...))

**Pull request merged**